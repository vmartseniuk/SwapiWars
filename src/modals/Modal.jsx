/* eslint-disable react/prop-types */
import React from "react";
import { makeStyles, Modal, Button } from "@material-ui/core";
import Backdrop from "@material-ui/core/Backdrop";
import Fade from "@material-ui/core/Fade";
import CloseIcon from "@material-ui/icons/Close";
import colors from "config/assets";

const useStyles = makeStyles((theme) => ({
  modal: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
  },
  container: {
    width: "100%",
    maxWidth: 800,
    backgroundColor: colors.cardLabelBg,
    borderRadius: "20px",
    position: "relative",
    outline: "none",
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    textAlign: "center",
    padding: "5rem 3.2rem 3.2rem",
    [theme.breakpoints.down("xs")]: {
      height: "100%",
      padding: "1.8rem 1.2rem 2.4rem",
    },
    [theme.breakpoints.down("sm")]: {
      margin: "5%",
      overflow: "auto",
      boxSizing: "border-box",
      maxWidth: "none",
    },
  },
  close: {
    position: "absolute",
    top: 0,
    right: 0,
    padding: ".5rem",
    borderRadius: "25px",
    [theme.breakpoints.down("xs")]: {
      position: "fixed",
    },
  },
}));

const CustomModal = (props) => {
  const { open, onClose, children } = props;
  const classes = useStyles();

  return (
    <Modal
      open={open}
      onClose={onClose}
      closeAfterTransition
      BackdropComponent={Backdrop}
      BackdropProps={{
        timeout: 500,
      }}
      className={classes.modal}
    >
      <Fade in={open}>
        <div className={classes.container}>
          <Button type="button" className={classes.close} onClick={onClose}>
            <CloseIcon />
          </Button>
          {children}
        </div>
      </Fade>
    </Modal>
  );
};

export default CustomModal;
