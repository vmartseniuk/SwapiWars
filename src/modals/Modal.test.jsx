import { create } from "react-test-renderer";
import Modal from "./Modal";

describe("FilmModal/Actors", () => {
  it("renders correctly", () => {
    const customComponent = (
      <div>
        <span>span1</span>
        <span>span2</span>
      </div>
    );
    const rendered = create(
      <Modal open={false} onClose={() => null}>
        {customComponent}
      </Modal>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
