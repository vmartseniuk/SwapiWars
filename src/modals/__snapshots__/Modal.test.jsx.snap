// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmModal/Actors renders correctly 1`] = `null`;
